pkgs/applications/science/programming/fdr/default.nix

```diff
@@ -0,0 +1,87 @@
+  qt57.qtbase
+  qt57.qtx11extras
+  ncurses
+  xorg.libX11
+  xorg.libXft
+  zlib
+  ];
+in
+stdenv.mkDerivation {
+  name = "fdr-4.2.0";
+  src = fetchurl {
+    url = https://www.cs.ox.ac.uk/projects/fdr/downloads/fdr-3754-linux-x86_64.tar.gz;
+    sha256 = "d24492485db9b8b95c62c53a6396094f836ee079cfc743688a397503c3ec9bf8";
+  };
+
+  phases = [ "unpackPhase" "installPhase" ];
+  installPhase = ''
+    mkdir -p "$out"
+
+    # shipped Qt is buggy
+    rm lib/libQt*
+    rm -r lib/qt_plugins
+
+    cp -r * "$out"
+    # Hack around lack of libtinfo in NixOS
+    ln -s ${ncurses.out}/lib/libncursesw.so.6 $out/lib/libtinfo.so.5
+    ln -s ${qt57.qtbase.out}/lib/qt5/plugins $out/lib/qt_plugins
+    ln -s ${zlib.out}/lib/libz.so.1 $out/lib/libz.so.1
+
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" "$out/bin/fdr4"
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/bin/_fdr4"
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" "$out/bin/refines"
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/bin/_refines"
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" "$out/bin/cspmprofiler"
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" "$out/bin/cspmexplorerprof"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/corei7/librefines.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/libcspm_process_compiler.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/libcsp_operators.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/_fdr.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/libfdr.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/libfdr_java.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/libprocess_compiler.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/librefines_gui.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/librefines_licensing.so"
+    patchelf --set-rpath "${stdenv.lib.makeLibraryPath (libs ++ [ "$out" ])}" \
+      "$out/lib/libboost_date_time.so.1.60.0"
```

Comment:
We have boost 1.60 in nixpkgs. Can't you use that instead of then delete this version?